### PR TITLE
Fix keypin column in commercial valuation data

### PR DESCRIPTION
**Pull request by wrridgeway**
While adding commercial data documentation to the wiki I was trying to determine if there is a way to define primary keys for the data (I couldn't). I did, however, determine that there were some pretty bad values for `keypin` in the data. We try to clean this data as little as possible, but given the importance of `keypin`, it feels prudent to make sure we don't have rows with random notes or "TOTAL PINS" as `keypin`.

This PR reduces the size of the dataset by about 500 rows and cleans they `keypin` column in about 1000 more.

### CODE REVIEW

File: etl/scripts-ccao-data-warehouse-us-east-1/ccao/ccao-commercial_valuation.R [hunk 1]
```diff
@@ -141,10 +141,28 @@ list.files(
...
+      paste(
+        substr(keypin, 1, 2),
+        substr(keypin, 3, 4),
+        substr(keypin, 5, 7),
+        substr(keypin, 8, 10),
+        substr(keypin, 11, 14),
+        sep = "-"
+      ),
+      keypin
+    ),
```
Comment (wrridgeway) [inline etl/scripts-ccao-data-warehouse-us-east-1/ccao/ccao-commercial_valuation.R:165]: I could not for the life of me get `ccao:pin_format_pretty` to work here.

File: etl/scripts-ccao-data-warehouse-us-east-1/ccao/ccao-commercial_valuation.R [hunk 2]
```diff
@@ -194,5 +212,6 @@ list.files(
   select(all_of(sort(names(.)))) %>%
   relocate(c(keypin, pins, township, year)) %>%
   relocate(c(file, sheet), .after = last_col()) %>%
+  distinct() %>%
```
Comment (wrridgeway) [inline etl/scripts-ccao-data-warehouse-us-east-1/ccao/ccao-commercial_valuation.R:215]: Oddly, there are some rows that are duplicated across all columns.

File: etl/scripts-ccao-data-warehouse-us-east-1/ccao/ccao-commercial_valuation.R [hunk 3]
```diff
@@ -141,10 +141,28 @@ list.files(
       )
   }, .progress = TRUE) %>%
   bind_rows() %>%
-  filter(check.numeric(excesslandval)) %>%
+  filter(
+    check.numeric(excesslandval),
+    !is.na(keypin),
+    !str_detect(keypin, "[:alpha:]"),
```
[the remark this answers is not in the document]
Comment (wrridgeway) [reply etl/scripts-ccao-data-warehouse-us-east-1/ccao/ccao-commercial_valuation.R:147]: No, key PINs should just be standard 14 digit PINs

File: etl/scripts-ccao-data-warehouse-us-east-1/ccao/ccao-commercial_valuation.R [hunk 4]
```diff
@@ -141,10 +141,21 @@ list.files(
...
+  ) %>%
   select(where(~ !all(is.na(.x)))) %>%
   # Add useful information to output and clean-up columns ----
   mutate(
+    keypin = str_trim(keypin),
+    keypin = str_pad(keypin, side = "left", width = 14, pad = "0"),
+    keypin = map(keypin, \(x) {
+      ifelse(nchar(x) == 14, pin_format_pretty(x, full_length = TRUE), x)
+    }),
+    keypin = str_pad(keypin, side = "right", width = 18, pad = "0")),
```
Comment (wrridgeway) [inline etl/scripts-ccao-data-warehouse-us-east-1/ccao/ccao-commercial_valuation.R:158]: This is a bit gross, but so is the data.

**Review by dfsnow — APPROVED** [review]
Alright, works for me! 👍

**Pull request merged**